samples/net/sockets/packet/src/packet.c

```diff
@@ -186,6 +190,12 @@ static int send_packet_socket(struct packet_data *packet)
 			}
 		}
 
+		/* If we have received any data, flush it here in order to
+		 * not to leak memory in IP stack.
+		 */
+		(void)recv(packet->send_sock, recv_buffer, sizeof(recv_buffer),
+			   0);
```

Comment:
`MSG_DONTWAIT` could be passed instead of using `fcntl`, so this patch/code would be shorter. The more important thing is that `MSG_DONTWAIT`  would affect only `recv()` operation and keep `send()` blocking, as previously. So choosing `fcntl` vs `MSG_DONTWAIT` depends on desired behavior.

Reply:
> `MSG_DONTWAIT` could be passed instead of using `fcntl`

Sure, good point.